common/src/main/java/com/wynntils/core/chat/tabs/ChatTab.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.chat.tabs;
+
+import com.wynntils.core.chat.RecipientType;
+import com.wynntils.wynn.event.ChatMessageReceivedEvent;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class ChatTab {
+    private String name;
+    private boolean lowPriority;
+
+    // Filters, inactive one is null
+    private Set<RecipientType> filteredTypes;
+    private String customRegexString;
+
+    private transient Pattern customRegex;
+
+    public ChatTab(String name, boolean lowPriority, Set<RecipientType> filteredTypes, String customRegexString) {
+        this.name = name;
+        this.lowPriority = lowPriority;
+        this.filteredTypes = filteredTypes;
+        this.customRegexString = customRegexString;
+
+        if (customRegexString != null) {
+            this.customRegex = Pattern.compile(customRegexString);
+        }
+    }
+
+    public boolean matchMessageFromEvent(ChatMessageReceivedEvent event) {
+        if (customRegexString != null) {
```

Comment:
I think this logic could be improved. Your comment above states ` // Filters, inactive one is null` but that is not really how this is implemented. I think the filters should be ANDed, and that `null` filters should count as `true` (or a match). 

Right now, if you have both a regexp and a recipient filter, and the regexp match, you will not check the recipient type, but return true regardless. And if you pass in `null` to both filters, you will get `false`. 

I think the first example should return true iff both the regexp and the recipient filter match. And I think the latter example should return `true` (and thus making the "All" ChatTab just use two null filters).